sunpy/map/tests/test_header_helper.py

```diff
@@ -321,3 +322,40 @@ def test_make_hpr_header(hgs_coord):
     assert header['hgln_obs'] == -50.0
     assert header['hglt_obs'] == 50.0
     assert Time(header['date-obs']) == hgs_coord.obstime
+
+def test_earth_observer_obsgeo_keywords():
+    """Test that ground-based observers get OBSGEO keywords."""
+    greenwich = EarthLocation(lat=51.4769*u.deg, lon=0*u.deg, height=46*u.m)
+
+    # Simple coordinate without observer in frame
+    coord = SkyCoord(0*u.arcsec, 0*u.arcsec,
+                     obstime="2020-01-01T12:00:00",
+                     frame=frames.Helioprojective)
+
+    data = np.zeros((512, 512))
+    header = make_fitswcs_header(data, coord, observer=greenwich)
+
+    assert 'obsgeo-x' in header
+    assert 'obsgeo-y' in header
+    assert 'obsgeo-z' in header
+    assert 'rsun_obs' in header
+
+def test_earth_observer_with_coordinate_observer_conflict():
+    """Test that explicit observer parameter takes precedence with warning."""
+    greenwich = EarthLocation(lat=51.4769*u.deg, lon=0*u.deg, height=46*u.m)
+
+    coord_with_observer = SkyCoord(0*u.arcsec, 0*u.arcsec,
+                                   obstime="2020-01-01T12:00:00",
+                                   observer='earth',
+                                   frame=frames.Helioprojective)
+
+    data = np.zeros((512, 512))
+
+    # Conflict case.
+    with pytest.warns(SunpyUserWarning, match="coordinate.observer.*ignoring"):
+        header = make_fitswcs_header(data, coord_with_observer, observer=greenwich)
+
+    assert 'obsgeo-x' in header
+    assert 'obsgeo-y' in header
+    assert 'obsgeo-z' in header
+    assert 'rsun_obs' in header
```

Comment:
You should add asserts here that `HGLN-OBS` and `HGLT-OBS` are not in the header.